lib/health_checks/sidekiq.rb

```diff
@@ -1,68 +1,60 @@
 require 'benchmark'
-require 'socket'
+require 'fileutils'
 require 'health_checks/checks/memory_check'
 require 'health_checks/checks/mongoid_check'
 require 'health_checks/checks/redis_check'
 
 module HealthChecks
   module_function
 
-  def sidekiq(config, mongo_databases, redis_configs)
+  def sidekiq(config, mongo_databases, redis_configs, sleep_seconds: 10)
     config.on(:startup) do
-      LivenessServer.new.start(mongo_databases, redis_configs)
+      LivenessThreadCheck.new.start(mongo_databases, redis_configs, sleep_seconds)
     end
   end
 
   private
 
-  class LivenessServer
-    LIVENESS_PORT = 8080
+  STATUS_FILE = '/tmp/sidekiq_ok'
 
-    def start(mongo_databases, redis_configs)
-      Sidekiq::Logging.logger.info "Starting liveness server on #{LIVENESS_PORT}"
+  class LivenessThreadCheck
+    def start(mongo_databases, redis_configs, sleep_seconds)
+      logger = Sidekiq::Logging.logger
 
-      Thread.start do
-        checks = mongo_databases.map { |db| Checks::MongoidCheck.new(db) }
-        checks += redis_configs.map{ |config| Checks::RedisCheck.new(config) }
-        checks << Checks::MemoryCheck.new
-        server = TCPServer.new(LIVENESS_PORT)
-        loop do
-          Thread.start(server.accept) do |socket|
-            elapsed_time = 0
-            begin
-              checks.each do |check|
-                elapsed_time = Benchmark.measure { check.run }
-                Sidekiq::Logging.logger.info "Time elapsed for #{check} was #{elapsed_time}"
-              end
-              respond_success(socket, 'Live!')
-            rescue => e
-              Sidekiq::Logging.logger.error "Health check failed with #{e.message}"
-              respond_failure(socket, e.message)
-            ensure
-              socket.close
-            end
-          end
+      logger.info "Starting liveness thread with #{sleep_seconds} sleep delay"
+
+      checks = mongo_databases.map { |db| Checks::MongoidCheck.new(db) }
+      checks += redis_configs.map{ |config| Checks::RedisCheck.new(config) }
+      # checks << Checks::MemoryCheck.new
+
+      Thread.new do
+        elapsed_time = 0
+
+        failed = false
+
+        checks.each do |check|
+          elapsed_time = Benchmark.measure { check.run }
+          logger.info "Time elapsed for #{check} was #{elapsed_time}"
+        rescue => e
+          logger.error e
+          logger.info "Time elapsed for #{check} was #{elapsed_time}"
+
+          failed = true
+          break
         end
-      end
-    end
 
-    def respond_success(socket, response)
-      respond(socket, '200 OK', response)
+        failed ? handle_fail : handle_success
+
+        sleep sleep_seconds
+      end
     end
 
-    def respond_failure(socket, response)
-      respond(socket, '500 Internal Server Error', response)
+    def handle_fail
+      FileUtils.rm(STATUS_FILE)
```

Comment:
This is not a reliable approach. If the whole process fails for some reason there will be noone to remove the file and you won't know something's wrong. Liveness checks should work by proving liveness, not trying to prove "death". For example you could write the last successful time that all liveness checks passed and compare that with the current time (with a margin) to see if it's been dead for a while.